src/item.cpp

```diff
@@ -3737,13 +3737,8 @@ bool item::goes_bad() const
     }
     if( is_corpse() ) {
         // Corpses rot only if they are made of rotting materials
-        std::vector<std::string> rotting_materials = {"Insect Flesh", "Human Flesh", "Flesh", "Bone"};
-        for( std::string material : rotting_materials ) {
-            if( made_of_types()[0]->name() == material ) {
-                return true;
-            }
-        }
-        return false;
+        static const std::set<material_id> rotting_materials = materials::get_rotting();
```

Comment:
This is a static variable, so it's initialized when the line if first run. On any successive runs it will not update.

If a new game is started (without quitting the program between) with mods that disable rotting for some material types, it won't update. Same if a new game starts with additional rotting materials.

It is easy to reproduce:
- start a game, spawn corpse that rots, save and go back to the main menu (leave the program running),
- change all `"rotting": true,` in the "materials.json" to ` "rotting": false,`
- start a new game or load the existing one.

Observe the corpses are still rotting.

Now restart the program and load the save and see that corpses are not rotting anymore.